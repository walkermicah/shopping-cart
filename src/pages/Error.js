import { useRouteError, useNavigate } from 'react-router-dom';
import '../styles/pages/Error.scss';

function Error() {
  const error = useRouteError();
  const navigate = useNavigate();

  return (
    <div className="Error">
      <div className="Error-message">
        <h3>Error!</h3>
        <p>Page could not be found.</p>
        <p>{error.message}</p>
        <button onClick={() => navigate(-1)}>Go back</button>
      </div>
    </div>
  );
}

export default Error;
