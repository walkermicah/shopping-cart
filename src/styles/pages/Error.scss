@import '../abstracts/mixins';

.Error {
  @include background-img;
  height: 100vh;
  padding: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: 'Lato', sans-serif;

  &-message {
    @include page-content;
    padding: 3rem;
    color: white;
    margin-bottom: 15rem;
    text-align: center;
    display: grid;
    gap: 2rem;
  }

  h3 {
    font-size: 5rem;
  }

  p {
    font-size: 2rem;
    font-weight: 300;
  }

  button {
    justify-self: center;
    background-color: black;
    color: white;
    border: 1px solid white;
    padding: 1rem 2rem;
    border-radius: 3px;
    font-size: 1.5rem;
    cursor: pointer;
    transition: transform 0.2s;
  }

  button:hover {
    transform: translateY(-1px);
  }
}
