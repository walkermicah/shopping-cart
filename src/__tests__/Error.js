import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  RouterProvider,
  createMemoryRouter,
  createRoutesFromElements,
  Route,
} from 'react-router-dom';
import Error from '../pages/Error';

const mockUseRouteError = jest.fn();
const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useRouteError: () => mockUseRouteError,
  useNavigate: () => mockUseNavigate,
}));

describe('Error Page Component', () => {
  let router;

  beforeEach(() => {
    router = createMemoryRouter(
      createRoutesFromElements(<Route path="/" element={<Error />} />)
    );
  });

  it('Renders error page', () => {
    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
  });

  it('Calls useNavigate when back button is clicked', () => {
    render(<RouterProvider router={router} />);
    const backBtn = screen.getByRole('button', {
      name: 'Go back',
    });
    userEvent.click(backBtn);
    expect(mockUseNavigate).toHaveBeenCalled();
    expect(mockUseNavigate).toHaveBeenCalledWith(-1);
  });
});
