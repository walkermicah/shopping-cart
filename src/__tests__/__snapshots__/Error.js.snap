// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page Component Renders error page 1`] = `
<div>
  <div
    class="Error"
  >
    <div
      class="Error-message"
    >
      <h3>
        Error!
      </h3>
      <p>
        Page could not be found.
      </p>
      <p />
      <button>
        Go back
      </button>
    </div>
  </div>
</div>
`;
